utility/src/main/java/org/oppia/android/util/parser/UrlImageParser.kt

```diff
@@ -41,24 +41,29 @@ class UrlImageParser private constructor(
   override fun getDrawable(urlString: String): Drawable {
     val imageUrl = String.format(imageDownloadUrlTemplate, entityType, entityId, urlString)
     val urlDrawable = UrlDrawable()
-    // TODO(#1039): Introduce custom type OppiaImage for rendering Bitmap and Svg.
+
     if (imageUrl.endsWith("svg", ignoreCase = true)) {
-      val target = SvgTarget(urlDrawable)
-      imageLoader.loadSvg("$gcsPrefix/$gcsResourceName/$imageUrl", CustomImageTarget(target))
+      val oppiaImage = OppiaImage.SvgImage(urlDrawable)
+      imageLoader.loadOppiaImage("$gcsPrefix/$gcsResourceName/$imageUrl",
+        findImageType(urlDrawable, oppiaImage))
     } else {
-      val target = BitmapTarget(urlDrawable)
-      imageLoader.loadBitmap("$gcsPrefix/$gcsResourceName/$imageUrl", CustomImageTarget(target))
+      val oppiaImage = OppiaImage.BitmapImage(urlDrawable)
+      imageLoader.loadOppiaImage("$gcsPrefix/$gcsResourceName/$imageUrl",
+        findImageType(urlDrawable, oppiaImage))
     }
     return urlDrawable
   }
 
-  private inner class BitmapTarget(urlDrawable: UrlDrawable) : CustomImageTarget<Bitmap>(
-    urlDrawable, { resource -> BitmapDrawable(context.resources, resource) }
-  )
-
-  private inner class SvgTarget(urlDrawable: UrlDrawable) : CustomImageTarget<PictureDrawable>(
-    urlDrawable, { it }
-  )
+  private fun findImageType(urlDrawable: UrlDrawable, oppiaImage: OppiaImage): CustomImageTarget {
+    return  when(oppiaImage) {
```

Comment:
I am stuck here, what should be the type Argument here for `CustomImageTarget`

Reply:
@BenHenning Can you please help me here?

Reply:
I think you want the target to be an ``OppiaImage`` but you need to update the Glide pipeline to support OppiaImage as a valid image type so that it knows how to bind it to a target (e.g. by extracting its bitmap or picture).

The idea behind ``OppiaImage`` is that it's self-service: you just load an Oppia image and bind it to a ``CustomImageTarget`` and Oppia's Glide integration takes care of the rest.

You can take a look at how we set up the custom support for Picture today--Glide doesn't support SVGs out of the box.

Reply:
@BenHenning I need a bit more help here. How could I update Glide pipeline to support our OppiaImage?

Reply:
@prayutsu take a look at RepositoryGlideModule. It registers a couple of custom type translators: SVG -> PictureDrawable, and InputStream -> SVG. We select this custom pipeline in GlideImageLoader by asking Glide to load a 'PictureDrawable' and it knows how to map from PictureDrawable to SVG to InputStream (which it will retrieve & provide for us). My suggestion is investigating whether we could set up a similar pipeline from Bitmaps/Pictures to the custom Oppia image type so that we can load that from Glide directly.

Note that we would probably still specialize between SVG & Bitmap, e.g.:

```kotlin
    Glide.with(context)
      .`as`(OppiaImage.BitmapImage::class.java)
      .load(model)
      .intoTarget(target)

    Glide.with(context)
      .`as`(OppiaImage.SvgImage::class.java)
      .fitCenter()
      .apply(RequestOptions.diskCacheStrategyOf(DiskCacheStrategy.NONE))
      .load(model)
      .intoTarget(target)
```

Does that help clarify?